server/routes/appointments/appointmentsController.ts

```diff
@@ -1237,13 +1276,14 @@ export default class AppointmentsController {
 
   private async getActionPlanAppointmentFromDraftOrService(
     req: Request,
-    res: Response
+    res: Response,
+    referralId?: string
   ): Promise<ActionPlanAppointment | null> {
     const { user } = res.locals
     const { accessToken } = user.token
     const { draftBookingId, sessionNumber, actionPlanId } = req.params
     if (draftBookingId) {
-      const fetchResult = await this.fetchDraftAppointmentOrRenderMessage(req, res)
```

Comment:
pass referral id through to draft process, so that back link is populated correctly